File: PG_Webhook_Signature_Verification_V1.py
import hmac
import hashlib
import base64

# Use signatureData.replace(" ","") if Payload contains white space

webhook_signature = "pgfi+UQVCpRBcB/BhlVxB26avdtFthcjm6qd3gDgMWc=" #signature recieved from Cashfree
#Replace your timestamp and payload below for "SignatureData" variable
signatureData = '1655711196296{"data":{"order":{"order_id":"order_1418522ApelfR0eqIOuEHR4pTaGS0FlYK","order_amount":1.00,"order_currency":"INR","order_tags":{"newKey-3":"New Value","newKey-2":"New Value","newKey":"New Value","newKey-1":"New Value"}},"payment":{"cf_payment_id":885299238,"payment_status":"SUCCESS","payment_amount":1.00,"payment_currency":"INR","payment_message":"Transaction pending","payment_time":"2022-06-20T13:16:33+05:30","bank_reference":"369852","auth_id":null,"payment_method":{},"payment_group":"wallet"},"customer_details":{"customer_name":null,"customer_id":"dummy11","customer_email":"ndfkn.ksdjk@cashfree.com","customer_phone":"9878685827"}},"event_time":"2022-06-20T13:16:36+05:30","type":"PAYMENT_SUCCESS_WEBHOOK"}'
message = bytes(signatureData, 'utf-8')
dig = hmac.new(b'TEST8b46e249a288e8727792ca129a331e24efbd91ed', msg=message, digestmod=hashlib.sha256).digest()#Replace your Secret_Key here
computed_siganture=base64.b64encode(dig).decode()

print("Webhook_Signature  :"+webhook_signature +"\n" + "Computed_Signature :" + computed_siganture)
if(computed_siganture==webhook_signature):
    print("WEBHOOK_VALIDATION_SUCCESS")
else:
    print("WEBHOOK_VALIDATION_FAILED")

File: PG_Webhook_Signature_Verification_V1.java
package JAVA;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;
import javax.crypto.Mac;
import javax.crypto.spec.SecretKeySpec;

public class PG_Webhook_Signature_Verification_V1 {
public static void main(String[] args) throws NoSuchAlgorithmException, InvalidKeyException
{
	String webhook_signature ="ZIiZUQ8mb0cud4NQaz8SoPKHX4cAZTBkvg5t8GBPLnI=";
	//data = timestamp+payload;
	String data = "1655496866280{\"data\":{\"order\":{\"order_id\":\"OD72237935\",\"order_amount\":40868.00,\"order_currency\":\"INR\",\"order_tags\":null},\"payment\":{\"cf_payment_id\":885296848,\"payment_status\":\"SUCCESS\",\"payment_amount\":40868.00,\"payment_currency\":\"INR\",\"payment_message\":\"Transaction pending\",\"payment_time\":\"2022-06-18T01:44:16+05:30\",\"bank_reference\":\"95940\",\"auth_id\":null,\"payment_method\":{\"card\":{\"channel\":null,\"card_number\":\"444433XXXXXX1111\",\"card_network\":\"visa\",\"card_type\":\"credit_card\",\"card_country\":\"IN\",\"card_bank_name\":\"Others\"}},\"payment_group\":\"credit_card\"},\"customer_details\":{\"customer_name\":\"sam\",\"customer_id\":\"12\",\"customer_email\":\"sam@gmail.com\",\"customer_phone\":\"9555595555\"}},\"event_time\":\"2022-06-18T01:44:26+05:30\",\"type\":\"PAYMENT_SUCCESS_WEBHOOK\"}";
	
	System.out.println(data);
	
	String secretKey = "TEST57d336883ec6e3bc3d86bc58dd0114a76bbbaac0"; // Get secret key from Cashfree Merchant Dashboard	;
	Mac sha256_HMAC = Mac.getInstance("HmacSHA256");
	SecretKeySpec secret_key_spec = new SecretKeySpec(secretKey.getBytes(),"HmacSHA256");
	sha256_HMAC.init(secret_key_spec);
	String computed_signature = Base64.getEncoder().encodeToString(sha256_HMAC.doFinal(data.getBytes()));
	System.out.println("Computed_Signature : "+computed_signature+"\n"+"Webhook_Signature : "+webhook_signature);
	if(webhook_signature.equals(computed_signature))
	{
		System.out.println("WEBHOOK SIGNATURE VALIDATION SUCCESS");
	}
	else
	{
		System.out.println("WEBHOOK SIGNATURE VALIDATION FAILED");	
	}
}
}
